File: src/wpsr/engines/types/RDFGeneric.java
package wpsr.engines.types;

import org.apache.uima.jcas.JCas;
import org.apache.uima.jcas.JCasRegistry;
import org.apache.uima.jcas.cas.TOP_Type;
import org.apache.uima.jcas.tcas.Annotation;

public class RDFGeneric extends Annotation {

    public final static int typeIndexID = JCasRegistry.register(RDFGeneric.class);
    public final static int type = typeIndexID;
    private String role = null;
    private double score = 0;

    @Override
    public int getTypeIndexID() {
	return typeIndexID;
    }

    protected RDFGeneric() {
    }

    public RDFGeneric(int addr, TOP_Type type) {
	super(addr, type);
	readObject();
    }

    public RDFGeneric(JCas jcas) {
	super(jcas);
	readObject();
    }

    public RDFGeneric(JCas jcas, int begin, int end) {
	super(jcas, begin, end);
	readObject();
    }

    public RDFGeneric(JCas jcas, int begin, int end, double score, String role) {
        super(jcas, begin, end);
        this.score = score;
        this.role = role;
    }

    private void readObject() {
    }

    public double getScore() {
        return score;
    }

    public String getRole() {
        return role;
    }
}

File: src/wpsr/analysis/types/RDFGeneric.ts
//TODO: import org.apache.uima.jcas.JCas;
//TODO: import org.apache.uima.jcas.JCasRegistry;
//TODO: import org.apache.uima.jcas.cas.TOP_Type;
//TODO: import org.apache.uima.jcas.tcas.Annotation;

module wpsr.engines.types {
  export class RDFGeneric extends Annotation {
    static typeIndexID: number = JCasRegistry.register("RDFGeneric");
    static type: number = typeIndexID;
    private role: string = null;
    private score: number = 0;
    constructor();
    constructor(cas: JCas);
    constructor(addr: number, type: TOP_Type);
    constructor(cas: JCas, begin: number, end: number);
    constructor(cas: JCas, begin: number, end: number, score: number, role: string) {
      // TODO: implement all constructors
      super(cas, begin, end);
      //super(addr, type);
      this.score = score;
      this.role = role;
      this.readObject();
    }
    private readObject(): void {
      // do nothing
    }
    public getScore(): number {
      return this.score;
    }
    public getRole(): string {
      return this.role;
    }
    public getTypeIndexID(): number {
      return typeIndexID;
    }
  }
}
